ee/middleware/upstreambasicauth/model.go

```diff
@@ -0,0 +1,49 @@
+package upstreambasicauth
+
+import (
+	"time"
+
+	"github.com/TykTechnologies/tyk/apidef"
+
+	"github.com/TykTechnologies/tyk/apidef/oas"
+	"github.com/TykTechnologies/tyk/internal/model"
+)
+
+const (
+	// ExtensionTykStreaming is the OAS extension for Tyk streaming.
+	ExtensionTykStreaming = "x-tyk-streaming"
+	StreamGCInterval      = 1 * time.Minute
+)
+
+// BaseMiddleware is the subset of BaseMiddleware APIs that the middleware uses.
+type BaseMiddleware interface {
+	model.LoggerProvider
+}
+
+// Gateway is the subset of Gateway APIs that the middleware uses.
+type Gateway interface {
+	model.ConfigProvider
+	model.ReplaceTykVariables
+}
+
+// APISpec is a subset of gateway.APISpec for the values the middleware consumes.
+type APISpec struct {
```

Comment:
I don't hate it, but why not make a complete copy of the struct in `model.`?
And ideally have an `func(*APISpec)Model()*model.APISpec` to bridge the gap with a single type.

That gets rid of a bunch of parameters in NewAPISpec or even delete it. Data models shouldn't have constructors, service models should and for what it's worth this is data model.

Reply:
APISpec.MergedAPI() part of follow up proposal in #6672, resolving